test/src/code_field/split_highlight_lines_test.dart

```diff
@@ -38,58 +15,73 @@ public class MyClass {
   }
 }''',
     expected: '''
-{value: 'public '}
-{className: class, children: ({value: ''}, {className: keyword, children: ({value: 'class'})}, {value: ' '}, {className: title, children: ({value: 'MyClass'})}, {value: ' '})}
-{value: '{
-'}
-{value: '  public '}
-{className: keyword, children: ({value: 'void'})}
-{value: ' main() { '}
-{className: comment, children: ({value: '// readonly'})}
-{value: '
-'}
-{value: '  }
-'}
-{value: '}'}''',
+{value: 'public ', children: empty}
+{className: class}
+  {value: '', children: empty}
+  {className: keyword}
+    {value: 'class', children: empty}
+  {value: ' ', children: empty}
+  {className: title}
+    {value: 'MyClass', children: empty}
+  {value: ' ', children: empty}
+{value: '{\n', children: empty}
+{value: '  public ', children: empty}
+{className: keyword}
+  {value: 'void', children: empty}
+{value: ' main() { ', children: empty}
+{className: comment}
+  {value: '// readonly', children: empty}
+{value: '\n', children: empty}
+{value: '  }\n', children: empty}
+{value: '}', children: empty}
+''',
   ),
 
   'Named sections': _Example(
     text: '''
 class MyClass {
-  void readOnlyMethod() {// [START section1]
+  void someMethod() {// [START section1]
```

Comment:
You may keep it, but I use `method1` and `method2` for such cases. It is perfectly symmetric so no method is special. And in this case it also mirrors 'section1' and 'section2'.

Another good naming is `startTrailingEndTrailing` and `startSeparateEndTrailing` to emphasize what we test here, then section names should mirror that as well.

A good code is the one that raises least 'why?' questions.